common/djangoapps/third_party_auth/models.py

```diff
@@ -827,6 +849,38 @@ def get_setting(self, name):
             return other_settings[name]
         raise KeyError
 
+    def get_current_saml_configuration(self):
+        """
+        Get the current active SAMLConfiguration for this provider.
+        Falls back to the default configuration if none is set.
+
+        When ENABLE_SAML_CONFIG_SIGNAL_HANDLERS toggle is enabled, this method includes
+        custom attributes for observability. When disabled, it uses legacy behavior.
+        """
+        if self.saml_configuration:
+            if ENABLE_SAML_CONFIG_SIGNAL_HANDLERS.is_enabled():
+                # .. custom_attribute_name: saml_config.using
+                # .. custom_attribute_description: Describes which config is being used: direct (from db),
+                #     latest, or default.
+                set_custom_attribute('saml_config.using', 'direct')
+            return self.saml_configuration
+
+        # Fall back to default configuration
+        try:
+            default_config = SAMLConfiguration.current(self.site_id, 'default')
```

Comment:
Can this fail, or does it just return None? I think `SAMLConfiguration.current(self.site_id, 'default')` was used in the legacy implementation outside of a try block, correct?